include/ginkgo/core/solver/lower_trs.hpp

```diff
@@ -196,23 +91,23 @@ class LowerTrs : public EnableLinOp<LowerTrs<ValueType, IndexType>>,
     }
 
     /**
-     * Gets the right hand side of the linear system.
+     * Returns the preconditioner operator used by the solver.
      *
-     * @return the right hand side
+     * @return the preconditioner operator used by the solver
      */
-    std::shared_ptr<const matrix::Dense<ValueType>> get_rhs() const
+    std::shared_ptr<const LinOp> get_preconditioner() const override
     {
-        return b_;
+        return preconditioner_;
     }
 
     /**
-     * Returns the preconditioner operator used by the solver.
+     * Get the triangular solve struct
      *
-     * @return the preconditioner operator used by the solver
+     * @return the trs solve struct
      */
-    std::shared_ptr<const LinOp> get_preconditioner() const override
+    gko::solver::SolveStruct *get_solve_struct() const
     {
-        return preconditioner_;
+        return solve_struct_.get();
     }
```

Comment:
You need to test these two (`get_preconditioner` and `get_solve_struct`), although they are trivial.
See [coverity's report](https://codecov.io/gh/ginkgo-project/ginkgo/pull/336/src/include/ginkgo/core/solver/lower_trs.hpp).

You technically even need to test that when you do pass in a preconditioner the preconditioner is actually generated (you get a a non identity I guess?).

Reply:
I actually do test the second case [here](https://github.com/ginkgo-project/ginkgo/blob/develop/core/test/solver/lower_trs.cpp#L73). But I should add tests for the getters.